modules/extension/validation/src/main/java/org/geotools/validation/relate/RelateIntegrity.java

```diff
@@ -154,8 +154,8 @@ private boolean validateMultipleLayers(
                 }
             }
         } finally {
-            fr1.close();
-            fr2.close();
+            if (fr1 != null) fr1.close();
+            if (fr2 != null) fr2.close();
```

Comment:
Every time I look at validation code I cry, looks like fr2 is not being closed in the inner loop (or is relying not iterating to the end to close).

Reply:
One could question why we're still keeping it there, has been un-used and un-maintained for years now...